runtime/ts/dom-particle-base.ts

```diff
@@ -79,112 +91,118 @@ export class DomParticleBase extends Particle {
     }
     this.currentSlotName = undefined;
   }
+
   _getStateArgs() {
     return [];
   }
-  forceRenderTemplate(slotName) {
+
+  forceRenderTemplate(slotName: string) {
     this._slotByName.forEach((slot, name) => {
-      if (!slotName || (name == slotName)) {
+      if (!slotName || (name === slotName)) {
         slot.requestedContentTypes.add('template');
       }
     });
   }
-  fireEvent(slotName, {handler, data}) {
+
+  fireEvent(slotName: string, {handler, data}) {
     if (this[handler]) {
       this[handler]({data});
     }
   }
+
   setParticleDescription(pattern) {
     if (typeof pattern === 'string') {
       return super.setParticleDescription(pattern);
     }
     assert(!!pattern.template && !!pattern.model, 'Description pattern must either be string or have template and model');
     super.setDescriptionPattern('_template_', pattern.template);
     super.setDescriptionPattern('_model_', JSON.stringify(pattern.model));
+    return undefined;
   }
-  /** @method clearHandle(handleName)
+
+  /**
    * Remove entities from named handle.
    */
-  async clearHandle(handleName) {
-    await this.handles.get(handleName).clear();
-    /*
+  async clearHandle(handleName: string) {
     const handle = this.handles.get(handleName);
-    if (handle.clear) {
+    if (handle instanceof Variable || handle instanceof Collection) {
```

Comment:
If we are getting rid of the fallback operation, we should throw if the `instanceof` check fails (because the clear operation has failed at that point).

Reply:
Isn't this code testing for the existence of the clear method?  So it would not have thrown originally?

Reply:
The code that tested for `clear` also had an `else` clause to handle the altenative.

Reply:
The diff is pretty hard to read, but the code that you see there was commented out.  Original code looked like this:

```
  async clearHandle(handleName) {
    await this.handles.get(handleName).clear();
    /*
    const handle = this.handles.get(handleName);
    if (handle.clear) {
      const entities = await handle.toList();
      if (entities) {
        return Promise.all(entities.map(entity => handle.remove(entity)));
      }
    }
    */
  }
```